### PR TITLE
Reserve ports for portworx processes during install

### PR DESCRIPTION
**Pull request by piyush-nimbalkar**
PWX-5590

### CODE REVIEW

File: frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java [hunk 1]
```diff
@@ -44,6 +50,92 @@ private static SchedulerBuilder createSchedulerBuilder(File yamlSpecFile) throws
...
+        Long startPort;
+        try {
+            startPort = Long.valueOf(System.getenv("PORTWORX_START_PORT"));
+        } catch (NumberFormatException e) {
+            startPort = DEFAULT_START_PORT;
+        }
+
+        List<Long> ports = new ArrayList<>();
+        if (startPort == DEFAULT_START_PORT) {
+            for (long port = 9001; port <= 9004; port++) {
```
Comment (disrani-px) [inline frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java]: These numbers need to be constants. Just reserce all ports from the start_port to start_port+range, no need of the holes

File: frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java [hunk 2]
```diff
@@ -30,9 +34,11 @@ public static void main(String[] args) throws Exception {
...
         SchedulerConfig schedulerConfig = SchedulerConfig.fromEnv();
         RawServiceSpec rawServiceSpec = RawServiceSpec.newBuilder(yamlSpecFile).build();
-        SchedulerBuilder schedulerBuilder = DefaultScheduler.newBuilder(
-                DefaultServiceSpec.newGenerator(rawServiceSpec, schedulerConfig, yamlSpecFile.getParentFile()).build(),
-                schedulerConfig)
+        DefaultServiceSpec serviceSpec = DefaultServiceSpec.newGenerator(
+                rawServiceSpec, schedulerConfig, yamlSpecFile.getParentFile()).build();
+
+        SchedulerBuilder schedulerBuilder =
+                DefaultScheduler.newBuilder(setPortResources(serviceSpec), schedulerConfig)
```
Comment (disrani-px) [inline frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java:44]: Add TaskEnvCannotChange for port env too

**Comment by piyush-nimbalkar** [comment]
Updated the PR

File: frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java [hunk 3]
```diff
@@ -44,6 +52,89 @@ private static SchedulerBuilder createSchedulerBuilder(File yamlSpecFile) throws
...
+            startPort = DEFAULT_START_PORT;
+        }
+
+        List<Long> ports = new ArrayList<>();
+        if (startPort == DEFAULT_START_PORT) {
+            for (long port = DEFAULT_START_PORT; port <= DEFAULT_END_PORT; port++) {
+                ports.add(port);
+            }
+        } else {
+            for (long port = startPort; port <= startPort + 18; port++) {
```
Comment (disrani-px) [inline frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java]: That should be 21 right? Use a constant there. No need for if/else here. Start from start_port to start_port+range

Comment (piyush-nimbalkar) [reply frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java]: We have holes in the default range. We actually need 19 ports. -r parameter reserves only consecutive ports, hence the if..else

Comment (disrani-px) [reply frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java]: By default it resevres up to 9022 though

Comment (piyush-nimbalkar) [reply frameworks/portworx/src/main/java/com/mesosphere/sdk/portworx/scheduler/Main.java]: Made the port count as constant.

**Review by disrani-px — APPROVED** [review]

**Pull request merged**